### PR TITLE
FastGeert LogServer

### PR DESCRIPTION
**Pull request by whoarethebritons**
https://ocd.appscale.com:8080/job/Daily%20Build/2542/

### CODE REVIEW

File: AppServer/google/appengine/api/logservice/logservice_stub.py [hunk 1]
```diff
@@ -14,339 +14,163 @@
...
-import os
 import time
-import threading
+import os
+import socket
+import struct
+import logging
+import base64
+import sys
+from Queue import Queue, Empty
```
Comment (tmarballi) [inline AppServer/google/appengine/api/logservice/logservice_stub.py]: Could you please alphabetize these imports?

File: AppServer/google/appengine/tools/devappserver2/server.py [hunk 2]
```diff
@@ -34,11 +34,13 @@
 import urlparse
 import wsgiref.headers
 
+import capnp
```
Comment (tmarballi) [inline AppServer/google/appengine/tools/devappserver2/server.py]: Move this import at the top.

File: LogService/logquery.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/python
+import argparse
+import socket
+import struct
+import time
+import logging
```
Comment (tmarballi) [inline LogService/logquery.py]: More alphabetization of imports :)

File: LogService/logserver.py [hunk 4]
```diff
@@ -0,0 +1,380 @@
+import struct
+import os
+import re
+import time
+from cStringIO import StringIO
```
Comment (tmarballi) [inline LogService/logserver.py]: Group the imports and the 'from import's together in alphabetic order.

File: LogService/twisted/plugins/appscale_logserver_plugin.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import os
+
+from zope.interface import implementer
+
+from twisted.python import usage
+from twisted.plugin import IPlugin
+from twisted.application.service import IServiceMaker
+from twisted.application import internet, service
+
+from logserver import LogServerFactory
```
Comment (tmarballi) [inline LogService/twisted/plugins/appscale_logserver_plugin.py]: Same as above.

File: AppServer/google/appengine/tools/devappserver2/server.py [hunk 6]
```diff
@@ -34,11 +35,12 @@
 import urlparse
 import wsgiref.headers
 
+
 from google.appengine.api import api_base_pb
 from google.appengine.api import apiproxy_stub_map
 from google.appengine.api import appinfo
 from google.appengine.api import request_info
-from google.appengine.api.logservice import log_service_pb
+from google.appengine.api.logservice import logging_capnp
```
Comment (cdonati) [inline AppServer/google/appengine/tools/devappserver2/server.py]: Is this file missing from the branch?

Comment (FastGeert) [inline AppServer/google/appengine/tools/devappserver2/server.py]: I made it a symlink, but @whoarethebritons replaced that through copying the file from the ~/LogService directory in the setup procedure.

**Comment by whoarethebritons** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/2569/

File: AppServer/google/appengine/tools/devappserver2/server.py (hunk 6 above)
Comment (whoarethebritons) [inline AppServer/google/appengine/tools/devappserver2/server.py]: @cdonati the symlink didn't work well with our rsync_files. The build should copy that file over.

**Pull request merged**